dotnet/src/SemanticKernel.Abstractions/AI/ChatClient/ChatClientExtensions.cs

```diff
@@ -66,4 +66,19 @@ public static IChatCompletionService AsChatCompletionService(this IChatClient cl
 
         return client.GetService<ChatClientMetadata>()?.ModelId;
     }
+
+    /// <summary>
+    /// Creates a new <see cref="KernelFunctionInvokingChatClient"/> that supports <see cref="Kernel"/> for function invocation with a <see cref="IChatClient"/>.
+    /// </summary>
+    /// <param name="client">Target chat client service.</param>
+    /// <returns>Function invoking chat client.</returns>
+    [Experimental("SKEXP0001")]
+    public static IChatClient AsKernelFunctionInvokingChatClient(this IChatClient client)
```

Comment:
Should I think of this as being exactly like `client.AsBuilder().UseFunctionInvocation().Build()`, except without a dependency on Microsoft.Extensions.AI? Or is it more than that?

Reply:
For the moment it's exactly that, in the next phase I will be looking into how to integrate the `AutoInvocationFilter` with this client, if I'm able to use the current API's probably won't be creating any extra customization and "native" copy would suffice, or even better, if `Microsoft.Extensions.AI` could work for .net 8. that would be ideal.